lib/llm/src/http/client.rs

```diff
@@ -176,6 +220,13 @@ impl AsyncEngineContext for HttpRequestContext {
         // For HTTP clients, killed is the same as stopped
         self.cancel_token.cancelled().await;
     }
+
+    fn link_child(&self, child: Arc<dyn AsyncEngineContext>) {
+        self.child_context
```

Comment:
Why does `child_context` need to be locked? Do we call `link_child` from many threads at once?

Reply:
> Why does child_context need to be locked?

This is to protect the `Vec` from concurrent access.

> Do we call link_child from many threads at once?

We don't currently. I think in Python the GIL should prevent any real concurrent calls into link_child.

In Rust, in the future, we may have two child threads each making a new outgoing request and linking the outgoing context to the incoming parent context, so when the parent request is cancelled, both child requests are cancelled.

I don't think this particular struct implementation will be used for the multi-threaded case, but we don't distinguish between which struct is used beneath the `AsyncEngineContext` trait, I think it is safer to allow this use case at the beginning.